src/main/kotlin/game/domain/Car.kt

```diff
@@ -0,0 +1,28 @@
+package game.domain
+
+class Car private constructor(val position: Int = 0, val name: String) {
+
+    fun move(moveCondition: Int): Car {
+        return when (canMove(moveCondition)) {
+            true -> advance()
+            false -> this
+        }
```

Comment:
2개 이상 조건일때만 when을 사용하고 
하나의 분기라면 if를 사용하셔도 좋을거같아요! 
```suggestion
        return if (canMove(moveCondition)) advance() else this
```